README.md

```diff
@@ -510,6 +514,17 @@ Per request metrics are available via a `metrics` event.
 }
 ```
 
+#### client.on('after', function(req, res, err) {... });
+StringClient and JsonClient emit an `after` event after each completed request
+(successful or not). This can be useful for attaching after handlers to report
+metrics or other useful information about the client. When using this event,
+the timings and metrics information is available via the request getters,
+`req.getTimings()` and `req.getMetrics()`. Note that the low level HttpClient
+does not support this method, as that client is a thin wrapper over the http
+streams. As such, the user is in full control of the request lifecycle, and
+`after` would not have a clear meaning.
```

Comment:
It seems that emitting an `'after'` event from the HttpClient would make things more consistent, but I don't have a strong opinion if the intention is for the `HttpClient` class to mostly be used internally by the restify clients implementation and not by external consumers.

As for the meaning of an 'after' event for the HttpClient, it could just be that it would be emitted when all the response data has been read, similar to when the `'end'` event is emitted on the underlying http(s) request object?

Reply:
The HttpClient is exposed publicly, so unfortunately not just an implementation detail.

Making `after` synonymous with `end` is good way to think about it. What would the value of `err` be in that scenario? Since user land is consuming the streams directly, errors could be created in any which way, and it would not be propagated back through to the `after` event. `after` would only be able to propagate request level errors like connection/request timeouts, unless we were to add some new API (but then it all feels very odd). 

Is that a reasonable approach? This would provide _some_ consistency, with caveats.